data-node/src/main/java/org/graylog/datanode/opensearch/statemachine/OpensearchStateMachineProvider.java

```diff
@@ -18,13 +18,23 @@
 
 import jakarta.inject.Inject;
 import jakarta.inject.Provider;
+import org.graylog.datanode.opensearch.OpensearchProcess;
+import org.graylog.datanode.opensearch.statemachine.tracer.StateMachineTracer;
+import org.graylog.datanode.opensearch.statemachine.tracer.StateMachineTracerAggregator;
+
+import java.util.Set;
 
 public class OpensearchStateMachineProvider implements Provider<OpensearchStateMachine> {
     private final OpensearchStateMachine opensearchStateMachine;
 
     @Inject
-    public OpensearchStateMachineProvider() {
-        this.opensearchStateMachine = OpensearchStateMachine.createNew();
+    public OpensearchStateMachineProvider(Set<StateMachineTracer> tracer, OpensearchProcess process) {
+        this.opensearchStateMachine = OpensearchStateMachine.createNew(process);
+        StateMachineTracerAggregator aggregator = opensearchStateMachine.getTracerAggregator();
```

Comment:
What about inverting the logic here and implement an addTracer method in the OpensearchStateMachine, instead requesting the aggregator and then attaching tracers to it? Basically hiding the implementation detail that there is an aggregator.

Reply:
good idea. I will move it into the `createNew` method altogether and let the generation handle it